sklearn/ensemble/_bootstrap.py

```diff
@@ -0,0 +1,64 @@
+"""Utility function to get the number of bootstrap samples."""
+
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+from numbers import Integral
+from warnings import warn
+
+
+def _get_n_samples_bootstrap(n_samples, max_samples, sample_weight):
+    """
+    Get the number of samples in a bootstrap sample.
+
+    Parameters
+    ----------
+    n_samples : int
+        Number of samples in the dataset.
+
+    max_samples : None, int or float
+        The maximum number of samples to draw.
+
+        - If None, then draw `n_samples` samples.
+        - If int, then draw `max_samples` samples.
+        - If float, then draw `max_samples * n_samples` unweighted samples or
+          `max_samples * sample_weight.sum()` weighted samples.
+
+    sample_weight : array of shape (n_samples,) or None
+        Sample weights with frequency semantics when `max_samples` is explicitly
+        set to a float or integer value. When keeping the `max_samples=None` default
```

Comment:
Above it says: "If int, then draw `max_samples` samples.", which sounds like sample weights only accounted for when `max_samples` is set to float?